Explorer/Assets/Scripts/MVC/WindowStackManager.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace MVC
+{
+    public class WindowStackManager : IWindowsStackManager
+    {
+        internal readonly List<IController> popupStack = new ();
+        internal readonly List<IController> persistentStack = new ();
+        internal IController fullscreenController;
+        internal IController topController;
+
+        public IController TopMostPopup => popupStack.LastOrDefault();
+
+        public PopupPushInfo PushPopup(IController controller)
+        {
+            int orderInLayer = popupStack.Count * 2;
+            popupStack.Add(controller);
+
+            return new PopupPushInfo(
+                    new CanvasOrdering(CanvasOrdering.SortingLayer.Popup, orderInLayer),
+                    new CanvasOrdering(CanvasOrdering.SortingLayer.Popup, orderInLayer - 1),
+                    popupStack.Count >= 2 ? popupStack[^2] : null);
+        }
+
+        public FullscreenPushInfo PushFullscreen(IController controller)
+        {
+            fullscreenController = controller;
+            return new FullscreenPushInfo(popupStack, new CanvasOrdering(CanvasOrdering.SortingLayer.Fullscreen, 0));
+        }
+
+        public void PopFullscreen(IController controller) =>
+            fullscreenController = null;
+
+        public PersistentPushInfo PushPersistent(IController controller)
+        {
+            persistentStack.Add(controller);
+            return new PersistentPushInfo(new CanvasOrdering(CanvasOrdering.SortingLayer.Persistent, -20));
+        }
+
+        public TopPushInfo PushTop(IController controller)
+        {
+            topController = controller;
+            return new TopPushInfo(popupStack, fullscreenController, new CanvasOrdering(CanvasOrdering.SortingLayer.Top, 1));
+        }
+
+        public void PopTop(IController controller) =>
+            topController = null;
+
+        public PopupPopInfo PopPopup(IController controller)
+        {
+            popupStack.Remove(controller);
+            return new PopupPopInfo(
+                new CanvasOrdering(CanvasOrdering.SortingLayer.Popup, ((popupStack.Count - 1) * 2) - 1),
+                TopMostPopup);
+        }
+    }
+
+    public readonly struct PopupPushInfo

```

Comment:
Let's make all these clasess internal

Reply:
If it's internal it can be used in MVCManager